src/Convex.jl

```diff
@@ -57,18 +59,24 @@ export add_constraints!, maximize, minimize, Problem, satisfy, solve!
 
 # Module level globals
 
+const ALLOW_DCP_VIOLATIONS = Ref{Bool}(false)
+
 """
-    emit_dcp_warnings
+    allow_dcp_violations()
+    allow_dcp_violations(emit)
+
+Controls whether or not errors are emitted for when an expression fails to be
+of disciplined convex form. When called without an argument, returns the current setting
+for whether or not DCP errors are emitted. To turn errors off, call
 
-Controls whether or not warnings are emitted for when an expression fails to be
-of disciplined convex form. To turn warnings off, override the method via
+    Convex.allow_dcp_violations(true)
 
-    Convex.emit_dcp_warnings() = false
+to (globally) allow non-DCP problems to be solved. Note that if a problem is not DCP, the solution obtained by Convex.jl can be wildly incorrect (even if the solution status is `OPTIMAL`!).
```

Comment:
Is there a practical use-case for allowing this? cc @mlubin

Reply:
I remember someone using it for alternating minimization. It could be something where they were going to fix a variable but hadn’t yet and the error occurs too early (when you construct the expression not when you solve the problem). The early errors should give more useful  stacktraces though.

Reply:
actually something weird with the current place the errors are thrown is that it is only when you call `vexity`. This happens during problem solving so we at least always throw then. But it doesn't happen when you construct the expression, unless you `show` it, which does check the vexity (to print it). So I think this isn't ideal, since you end up with different-seeming behavior in the REPL (as it shows things).

Reply:
What about just erroring on `optimize!`?

Reply:
ok, I've made that changed, and removed the opt-out